packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -257,44 +308,88 @@ function MapAnalysisImpl(props: Props & CompleteAppState) {
-      icon: <Download />,
-    },
-  ].map((item, index) => {
-    return (
-      <button
-        style={buttonStyles}
-        onClick={() => {
-          setActiveSideMenuItems((currentSet) => {
-            const newSet = new Set(currentSet);
-            newSet.has(index) ? newSet.delete(index) : newSet.add(index);
-            return newSet;
-          });
-        }}
-      >
-        <span style={iconStyles} aria-hidden>
-          {item.icon}
-        </span>
-        <span style={labelStyles}>{item.labelText}</span>
-      </button>
+  type SideNavigationItemConfigurationObject = {
+    href?: string;
+    labelText: string;
+    icon: ReactNode;
+    renderWithApp: (app: App) => ReactNode;
+  };
+  const sideNavigationRenderPlaceholder: SideNavigationItemConfigurationObject['renderWithApp'] =
+    (_) => (
+      <div style={{ padding: '2rem' }}>
+        <p>Not Implemented!</p>
+      </div>
     );
-  });
+  const sideNavigationButtonConfigurationObjects: SideNavigationItemConfigurationObject[] =
+    [
+      {
+        labelText: 'Paint',
+        icon: <EditLocation />,
+        renderWithApp: sideNavigationRenderPlaceholder,
+      },
+      {
+        labelText: 'Filter',
+        icon: <Filter />,
+        renderWithApp: sideNavigationRenderPlaceholder,
+      },
+      {
+        labelText: 'Plot',
+        icon: <BarChartSharp />,
+        renderWithApp: (app) => {
+          return (
+            <FloatingVizManagement
+              analysisState={analysisState}
+              setActiveVisualizationId={setActiveVisualizationId}
+              appState={appState}
+              app={app}
+              geoConfigs={geoConfigs}
+              totalCounts={totalCounts}
+              filteredCounts={filteredCounts}
+              toggleStarredVariable={toggleStarredVariable}
+              filters={filtersIncludingViewport}
+            />
+          );
+        },
+      },
+      {
+        labelText: 'Download',
+        icon: <Download />,
+        renderWithApp: sideNavigationRenderPlaceholder,
+      },
+      {
+        labelText: 'Share',
+        icon: <Share />,
+        renderWithApp: sideNavigationRenderPlaceholder,
+      },
+      {
+        labelText: 'Save',
+        icon: <Save />,
+        renderWithApp: sideNavigationRenderPlaceholder,
+      },
+    ];
+
+  const [activeSideMenuIndex, setActiveSideMenuIndex] =
+    useState<number | undefined>();
+
+  const sideNavigationButtons = sideNavigationButtonConfigurationObjects.map(
+    ({ labelText, icon }, index) => {
+      return (
+        <button
+          style={buttonStyles}
+          onClick={() => {
+            setActiveSideMenuIndex((currentIndex) =>
+              currentIndex === index ? undefined : index
+            );
+          }}
+        >
+          <span style={iconStyles} aria-hidden>
+            {icon}
+          </span>
+          <span style={labelStyles}>{labelText}</span>
+        </button>
+      );
+    }
+  );
```

Comment:
**Note to benevolent reviewer(s)**: this and following lines are changes made to the configuration and rendering of the lefthand side navigation items.

Reply:
Resolved in: https://github.com/VEuPathDB/web-monorepo/pull/8/commits/c853e0b19b033d59d553093b24d545c941190ada